BaragonAgentService/src/main/java/com/hubspot/baragon/agent/managed/LifecycleHelper.java

```diff
@@ -246,6 +246,64 @@ public void applyCurrentConfigs() throws AgentServiceNotifyException {
     }
   }
 
+  public void applyAll(List<Optional<Pair<ServiceContext, Collection<BaragonConfigFile>>>> toApply) {
+    toApply.parallelStream().forEach(item -> {
+      try {
+        configHelper.bootstrapApply(item.get().getKey(), item.get().getValue());
+      } catch (Exception e) {
+        LOG.error(String.format("Caught exception while applying %s during bootstrap", item.get().getKey().getService().getServiceId()), e);
+      }
+    });
+  }
+
+  public void applyCurrentConfigs() throws AgentServiceNotifyException {
+    LOG.info("Getting current state of the world from Baragon Service...");
+
+    final Stopwatch stopwatch = Stopwatch.createStarted();
+    final long now = System.currentTimeMillis();
+
+    final Collection<String> services = stateDatastore.getServices();
+    if (services.size() > 0) {
+      ExecutorService executorService = Executors.newFixedThreadPool(services.size());
+      List<Callable<Optional<Pair<ServiceContext, Collection<BaragonConfigFile>>>>> todo = new ArrayList<>(services.size());
+
+      Optional<Integer> maybeVersion = stateDatastore.getStateVersion();
+      if (maybeVersion.isPresent()) {
+        bootstrapStateNodeVersion.set(maybeVersion.get());
+      }
+
+      for (BaragonServiceState serviceState : getGlobalStateWithRetry()) {
+        if (!(serviceState.getService().getLoadBalancerGroups() == null) && serviceState.getService().getLoadBalancerGroups().contains(configuration.getLoadBalancerConfiguration().getName())) {
+          todo.add(new BootstrapFileChecker(configHelper, serviceState, now));
+        }
+      }
+
+      LOG.info("Going to apply {} services...", todo.size());
+
+      try {
+        List<Optional<Pair<ServiceContext, Collection<BaragonConfigFile>>>> toApply = new ArrayList<>();
+        List<Future<Optional<Pair<ServiceContext, Collection<BaragonConfigFile>>>>> applied = executorService.invokeAll(todo);
+        for (Future<Optional<Pair<ServiceContext, Collection<BaragonConfigFile>>>> serviceFuture : applied) {
+          Optional<Pair<ServiceContext, Collection<BaragonConfigFile>>> maybeToApply = serviceFuture.get();
+          if (maybeToApply.isPresent()) {
+            toApply.add(maybeToApply);
+          }
+        }
+        applyAll(toApply);
+        configHelper.checkAndReload();
+      } catch (Exception e) {
+        LOG.error("Caught exception while applying and parsing configs", e);
+        if (configuration.isExitOnStartupError()) {
+          Throwables.propagate(e);
```

Comment:
This is fine for the exitOnStartupErrorCase. But, if we are not in that case, we are now left with nginx in a state of invalid configuration (meaning it never actually loaded any of it into memory and stared serving for real). We need to figure out what we want to do in this case.
- Can we find if it's just one file? Maybe remove that individual one and keep going?
- If >x files are invalid should we exit and not even try to revert a smaller portion of them?